File: Digital Clock & Countdown Timer.py
import time
import sys
import select

def userChoice(choice):
    if choice == "1":
        digital_clock()
        
    elif choice == "2":
        seconds = int(input("Enter the number of seconds to countdown: "))
        countdown_timer(seconds)
        
    else:
        print("Invalid choice!")

def digital_clock():
    print("\n+-----------------------+")
    print("|  Current Time:        |")
    
    while True:
        current_time = time.localtime()
        time_string = time.strftime("%Y-%m-%d %H:%M:%S", current_time)
        
        # Clear the line for the time and print the updated time
        print(f"\r|  {time_string}  |", end="")        
        time.sleep(1)
        
        # Check for user input to quit
        if sys.stdin in select.select([sys.stdin], [], [], 0)[0]:
            user_input = sys.stdin.read(1).strip().lower()
            if user_input == 'q':
                break

    print("\n+-----------------------+")

def countdown_timer(seconds):
    while seconds > 0:
        print(f"\rTime remaining: {seconds} seconds", end="", flush = True)
        time.sleep(1)
        seconds -= 1
    print("Time's up!")

if __name__ == '__main__':
    while True:
        try:
            choice = input("Choose an option (1: Digital Clock, 2: Countdown Timer, q: Quit): ")
            if choice.lower() == 'q':
                break
            userChoice(choice)

        except EOFError:
            break

File: Digital Clock & Countdown Timer/Digital Clock & Countdown Timer.cpp
#include <iostream>
#include <iomanip>
#include <ctime>
#include <thread>
#include <chrono>
using namespace std;

// Function declarations
void userChoice(int choice);
void digital_clock();
void countdown_timer(int seconds);

// Function to accept the user's choice
void userChoice(int choice)
{
    if (choice == 1) 
    {
        digital_clock();
    }
    
    else if (choice == 2) 
    {
        int seconds;
        cout << "Enter the number of seconds to countdown: ";
        cin >> seconds;
        countdown_timer(seconds);
    } 
    
    else
    {
        cout << "Invalid choice!" << endl;
    }
}

// Function for displaying a digital clock
void digital_clock() 
{
    while (true) 
    {
        // Get the current time
        time_t now = time(nullptr);
        tm* local_time = localtime(&now);

        // Print the current time
        cout << "\rCurrent Time: " << put_time(local_time, "%H:%M:%S") << flush;

        // Pause for 1 second
        this_thread::sleep_for(chrono::seconds(1));
    }
}

// Function to display a countdown timer
void countdown_timer(int seconds) 
{
    cout << "Countdown Timer started!" << endl;
  
    while (seconds > 0) 
    {
        cout << "\rTime remaining: " << seconds << " seconds" << flush;
        this_thread::sleep_for(chrono::seconds(1));
        seconds--;
    }
  
    cout << "\nTime's up!" << endl;
}

// Driver code
int main() 
{
    while (true) 
    {
        int choice;
        cout << "Choose an option (1: Digital Clock, 2: Countdown Timer): ";
        cin >> choice;
        userChoice(choice);
    }

    return 0;
}
